tdms/tests/system/data/input_generation/matfile_option_edit.py

```diff
@@ -0,0 +1,99 @@
+import os
+from dataclasses import dataclass
+from pathlib import Path
+
+import numpy as np
+from hdf5storage import loadmat, savemat
+
+LOCATION_OF_THIS_FILE = os.path.abspath(os.path.dirname(__file__))
+
+
+@dataclass
+class OverwriteWith:
+    """Small container class that will be used when overwriting the "flag" variables usecd and intmethod when a config file specifies an 'adjust' field.
+
+    Intended workflow along the lines of:
+    intmethod = OverwriteWith(whether_to_overwrite, value_to_replace_with)
+    if intmethod.write:
+        h5data['intmethod'] = intmethod.value
+    else:
+        pass
+    """
+
+    # Flags whether or not to overwrite a value with self.value
+    write: bool
+    # The value to write if we are intending to replace a value in a file
+    value: bool | int | float | None
+
+
+def edit_mat_file(
+    test_directory: Path | str,
+    to_produce: Path | str,
+    options: dict[str, str | bool | int],
+) -> None:
+    """Produce a new .mat file by copying another .mat file and changing some of the TDMS "flag" variables according to the options provided.
+
+    This function can adjust the values of the following variables:
+    - usecd: Default value is 1 (FDTD). 0 results in use of PSTD
+    - intmethod: Default value is 1 (cubic). 2 results in BLI
+
+    :param test_directory: Directory in which the existing .mat file to use as a basis is stored.
+    :param to_produce: Filename to write the edited .mat file to.
+    :param options: Dictionary specifying the variables to adjust and the new values to assign to them.
+    """
+    # Get name of input file to produce immediately
+    to_produce = f"{str(test_directory)}/{str(to_produce)}"
+    # Read the name of the .mat file to adjust
+    adjust_file = f"{test_directory}/{options['adjust']}"
+
+    # Read the variable values that we want to override, provided they exist
+    # Check if we want to overwrite the solver method
+    if "solver_method" in options.keys():
```

Comment:
```suggestion
    if "solver_method" in options:
```
[`in`](https://docs.python.org/reference/expressions.html#membership-test-operations) already tests for the existence of a key in a [dict](https://docs.python.org/library/stdtypes.html#dict).

Also, "if solver method in options" is _aaaalamost_ plain English 👍